### PR TITLE
Pin down React version on v26

### PR DESCRIPTION
**Pull request by javache**
React 15.1 is not compatible with v26, so prevent people from getting a bad experience on a fresh react-native install.

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
By analyzing the blame information on this pull request, we identified @cpojer and @davidaurelio to be potential reviewers.

**Comment by javache** [comment]
Some of the problems in https://github.com/facebook/react-native/issues/7679 could be fixed by this.

**Comment by bestander** [comment]
This PR does not make sense, please read https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md

**Comment by javache** [comment]
😠

**Comment by q--** [comment]
A little more explanation of what's going on:

In spite of the fact that React is supposedly following SemVer, React@15.1.0 is not fully backwards compatible with React@15.0.2. Using React-Native@0.26.x with React@15.1.0 is required to work according to SemVer, but it is broken in practice.

Using React with these modules installed works:
```
npm ls --depth=0
MyApp@0.0.1 /home/username/Documents/MyApp
├── react@15.0.2
├── react-native@0.26.3
```
However, because in `package.json` `dependencies` is specified as follows:

    "dependencies": {
      "react": "^15.0.2",
      "react-native": "^0.26.1"
    }
running `npm update` will automatically upgrade React to version `15.1.0`. And if you run `react-native run-android` after upgrading, you will get the error

    undefined is not an object (evaluating 'ReactPerf().injection.injectMeasure')

or sometimes the error:

    Module AppRegistry is not a registered callable module
although the latter may have resulted from attempts to fix the problem.

(Another scenario where React gets upgraded to 15.1.0 is after removing the module using `npm uninstall` without `--save` or by removing the `node_modules` folder and then executing `npm install`.)

**TL;DR:** running `npm update` will upgrade React to a version that is incompatible with React Native, even though React must be backwards compatible according to [SemVer](http://semver.org/).

_(Note: React [did a rewrite](https://github.com/facebook/react/blob/master/CHANGELOG.md#1510-may-20-2016) of React Perf in 15.1.0)_

@bestander I hope I experienced the issue clearly enough. I think it's the same problem @javache is experiencing.

**Comment by bestander** [comment]
@q--, @javache is my colleague and a good friend of mine, I just was joking when wrote the message that his PR does not make sense.
Sorry for making you confused.
We actually made the fix in the next commit https://github.com/facebook/react-native/commit/f9fff0e26a58f30793cd0e85d41c55bbdb88f9db.

And thanks for articulating the problem so well, you are right, 15.1.0 is a breaking change and React probably should have been bumped to 16.0.0 in this case.
Although I understand the stress that React team has and a possible breaking change may slip out of visibility. They promised to have less breaking changes going forward.

As for what to do:
- Latest 0.26 already has strict version in package.json https://github.com/facebook/react-native/blob/0.26-stable/package.json
- The only place where I found `^` in react version was in a new project package.json that resulted with `react-native init` command. This has been fixed with release 0.26.3 yesterday.

0.27 should be compatible with React 15.1.0 but the problem of transitive dependencies introducing a breaking change is still relevant. 
We will keep working on finding a solution to ensure that dependencies don't get updated uncontrollably in the future.

Do you still have a scenario when a RN app can break because of React 15.1.0 update?

**Comment by q--** [comment]
@bestander No, I can't think of any other scenario where this will break once this ends up in a release. If the fix handles upgrading as well, that is.
